clients/include/testing_asum_strided_batched.hpp

```diff
@@ -98,29 +92,74 @@ hipblasStatus_t testing_asum_strided_batched(const Arguments& argus)
             return status_4;
     }
 
-    if(device_pointer)
-        CHECK_HIP_ERROR(hipMemcpy(
-            rocblas_result2, d_rocblas_result, sizeof(T2) * batch_count, hipMemcpyDeviceToHost));
+    CHECK_HIP_ERROR(hipMemcpy(
+        hipblas_result_device, d_hipblas_result, sizeof(Tr) * batch_count, hipMemcpyDeviceToHost));
 
-    if(argus.unit_check)
+    if(argus.unit_check || argus.norm_check)
     {
         /* =====================================================================
                     CPU BLAS
         =================================================================== */
         for(int b = 0; b < batch_count; b++)
         {
-            cblas_asum<T1, T2>(N, hx.data() + b * stridex, incx, &cpu_result[b]);
+            cblas_asum<T, Tr>(N, hx.data() + b * stridex, incx, &cpu_result[b]);
         }
 
         if(argus.unit_check)
         {
-            unit_check_general<T2>(1, batch_count, 1, cpu_result, rocblas_result1);
-            unit_check_general<T2>(1, batch_count, 1, cpu_result, rocblas_result2);
+            unit_check_general<Tr>(1, batch_count, 1, cpu_result, hipblas_result_host);
+            unit_check_general<Tr>(1, batch_count, 1, cpu_result, hipblas_result_device);
+        }
+        if(argus.norm_check)
+        {
+            hipblas_error_host
+                = norm_check_general<Tr>('F', 1, batch_count, 1, cpu_result, hipblas_result_host);
+            hipblas_error_device
+                = norm_check_general<Tr>('F', 1, batch_count, 1, cpu_result, hipblas_result_device);
         }
 
     } // end of if unit/norm check
 
-    //  BLAS_1_RESULT_PRINT
+    if(argus.timing)
+    {
+        hipStream_t stream;
+        status_1 = hipblasGetStream(handle, &stream);
+        status_2 = hipblasSetPointerMode(handle, HIPBLAS_POINTER_MODE_DEVICE);
+        if(status_1 != HIPBLAS_STATUS_SUCCESS || status_2 != HIPBLAS_STATUS_SUCCESS)
+        {
+            hipblasDestroy(handle);
```

Comment:
If you are updating a lot of these maybe you should create the hipBlasLocalHandle to just do the stack unwinding destroy call like rocblas ....  sure easy for me to say :-)

Reply:
Yeah I contemplated it but was too lazy this time, I'll do it next time I swear (🤞) . It would be nice to use a CHECK macro to get rid of a bunch of this code.